### PR TITLE
Update `dependencies.yaml` to support CUDA 12.*

### PR DESCRIPTION
**Pull request by pentschev**

### CODE REVIEW

File: conda/environments/all_cuda-120_arch-x86_64.yaml
Comment (ajschmidt8) [inline conda/environments/all_cuda-120_arch-x86_64.yaml:1]: Is there any reason why this file was removed? I would expect it to stick around due to this line in `dependencies.yaml`:

- https://github.com/rapidsai/ucxx/blob/c84c0f976c5b4b5b6d489366696013e77ed515b4/dependencies.yaml#L6

Typically this should be caught by pre-commit, but there is an issue with pre-commit identifying new files as described in this issue:

- https://github.com/pre-commit/pre-commit/issues/2789

**Review by quasiben — APPROVED** [review]

File: conda/environments/all_cuda-120_arch-x86_64.yaml
Comment (ajschmidt8) [reply conda/environments/all_cuda-120_arch-x86_64.yaml:1]: ok, i'll have to investigate this separately. i will approve for now.

**Review by ajschmidt8 — APPROVED** [review]

File: dependencies.yaml [hunk 1]
```diff
@@ -115,7 +115,7 @@ dependencies:
               - cuda-version=11.8
               - cudatoolkit
           - matrix:
-              cuda: "12.0"
+              cuda: "12.*"
             packages:
               - cuda-version=12.0
               - cuda-cudart-dev
```
Comment (bdice) [inline dependencies.yaml:121]: This isn't right. We shouldn't be installing `cuda-version=12.0` for all `cuda 12.*` inputs.

File: conda/environments/all_cuda-120_arch-x86_64.yaml
Comment (bdice) [reply conda/environments/all_cuda-120_arch-x86_64.yaml:1]: Running `pre-commit run --all-files` generated this file again for me. I don't think it was supposed to be removed.

Comment (pentschev) [reply conda/environments/all_cuda-120_arch-x86_64.yaml:1]: Indeed the file showed up locally now, see https://github.com/rapidsai/ucxx/pull/146#pullrequestreview-1802934101. It should be addressed in there.

**Pull request merged**